Print_A_Bot/djwifi/templates/djwifi/configure.html

```diff
@@ -0,0 +1,246 @@
+        </div>
+        <div class="modal-footer text-center">
+            <button type="button" class="btn btn-default" data-dismiss="modal">{% trans 'Close' %}</button>
+            <button type="submit" class="btn btn-primary">{% trans 'Save changes' %}</button>
+        </div>
+        <input type="hidden" name="wifi_ssid" value="{% verbatim %}{{ essid }}{% endverbatim %}">
+        {% csrf_token %}
+    </form>
+</script>
+
+<script id="template-wifi-submit" type="text/x-handlebars-template">
+    <div id="wifi-submit-wrapper" class="jumbotron text-center">
+        <h2>{% trans 'Garden Genie is Attempting to connect to' %} {% verbatim %}{{ essid }}{% endverbatim %}.</h2>
+
+        <p>
+            {% verbatim %}{{> waiting }}{% endverbatim %}
+        </p>
+
+        <p>{% trans 'When the light on your garden genie turns green you can check in through the cloud.' %}</p>
+        <p>{% trans 'Please re-connect to your regular wifi network and feel free to check back in with the Local button below.' %}</p>
+
+        <div class="row">
+            <div class="col-xs-6 col-sm-3 col-md-1 col-sm-offset-3 col-md-offset-5">
+                <a class="btn btn-success" href="http://gardengenie.local">{% trans 'Local' %}</a></div>
+            <div class="col-xs-6 col-sm-3 col-md-1">
+                <a class="btn btn-success" href="http://www.solargreens.net">{% trans 'Cloud' %}</a></div>
+        </div>
+    </div>
+</script>
+
+<script>
+    window.access_points = null;
+    window.access_point = null;
+
+    function configure_wifi(network_id) {
+        window.access_point = window.access_points[network_id];
+
+        console.log('access_point');
+        console.log(window.access_point);
+        if (window.access_point.encrypted) {
+            var template = Handlebars.compile($('#template-wifi-configure').html());
+            show_modal_content(template(window.access_point));
+
+            $('#show-password').change(function() {
+                var pass_type = ($(this).prop('checked')) ? 'text' : 'password';
+                $('#passkey').prop('type', pass_type);
+            });
+
+            $('#wifi-config-form').submit(function(e) {
+                e.preventDefault();
+
+                var wrapper = $('#modal-body');
+                var waiting_html = Handlebars.compile($('#template-waiting').html());
+                wrapper.html(waiting_html({msg: '{% trans 'Saving configuration' %}'}));
+
+                $.post('{% url 'wifi:connect' %}', $(this).serialize(), function(data) {
+                    $(modal_id).modal('hide');
+                    notify_data(data);
+                });
+            });
+        } else {
+            connect_wifi(network_id);
+        }
+    }
+
+    function connect_wifi(network_id) {
+        var template = Handlebars.compile($('#template-waiting').html());
+        show_modal_content(template({msg: '{% trans 'Connecting' %}...'}));
+        $.post('{% url 'wifi:connect' %}', {
```

Comment:
`$.post` is a jquery function to send ` POST` call via ajax

http://api.jquery.com/jQuery.post/

it's an older javascript framework but does s fair amount without getting to complex